features/GEM003_Unique-representation-identifier.feature

```diff
@@ -0,0 +1,18 @@
+@informal-proposition
+@GEM
+Feature: GEM003 - Unique Representation Identifier
+
+  # Scenario: No Shape Representation identifier shall be used twice within the product representation of an element 
+
+      # Given An IfcProductDefinitionShape
+      #   And The element has more than 1 instance(s) of IfcShapeRepresentation
+
+      #   Then Each instance of IfcShapeRepresentation has a unique value for the attribute RepresentationIdentifier
+
+  Scenario: No Shape Representation identifier shall be used twice within the product representation of an element 
+
+      Given An IfcProduct
+        And Its values for attribute Representation
+        And Its values for attribute Representations
+
+        Then The element must have unique values for the attribute RepresentationIdentifier
```

Comment:
I'm confused: 

1. `IfcProduct` does not have `Representations` attribute. What exactly is being checked here?
2. _Given an `IfcProduct`_, but _The **element** must have ..._ - provided, that IFC schema does have an `IfcElement`, should we perhaps choose a different word here to diminish the possibility of misunderstanding? Especially given the scenario title: _**product** representation of an **element**_.
3. `IfcProduct` does not have `RepresentationIdentifier` attribute. Similar question to 1..

Reply:
The following IA is checked: https://standards.buildingsmart.org/documents/Implementation/IFC_Implementation_Agreements/CV-2x3-171.html
 `Any product representation of an element may have multiple shape representations. However there shall be no two or more IfcShapeRepresentations having the same RepresentationIdentifier within one IfcProductDefinitionShape.`

There was a discussion about this (in https://github.com/buildingSMART/ifc-gherkin-rules/issues/22). In short, with making gherkin statements there is often a tradeoff between readability and reuse. A more understandable statement would be, for example
```
Given an IfcProduct
 Then Its representations should have unique identifiers
```
In Python code/IFC we would have to navigate from the IfcProduct (a wall, a column, etc) to the shape representations and then check if the IA is met. This works, but we will write it once and never use it again. Modular statements as the ones above lead to the same result (an IfcWall for example has a representation IfcProductDefinitionShape, these have representations as IfcShapeRepresentation and these will have values RepresentationIdentifier), but will be way more suitable for reuse as rules will probably often be some attributes being checked for a certain condition. This will then save development time and lead to cleaner code. 
The only thing is that we will have to add some additional comments/description for when the statements are not completely self-explanatory, which is the case for this rule as we see now ..